app/src/main/java/com/activity_sync/screens/SettingsScreen.java

```diff
@@ -0,0 +1,253 @@
+{
+    @Inject
+    INavigator navigator;
+
+    @Inject
+    IPermanentStorage permanentStorage;
+
+    @Bind(R.id.notification_switch)
+    SwitchCompat notificationSwitch;
+
+    @Bind(R.id.notification_sound_checkbox)
+    AppCompatCheckBox notificationSoundCheckBox;
+
+    @Bind(R.id.notification_vibrate_checkbox)
+    AppCompatCheckBox notificationVibrateCheckBox;
+
+    @Bind(R.id.location_switch)
+    SwitchCompat locationSwitch;
+
+    @Bind(R.id.days_seekbar)
+    AppCompatSeekBar daysSeekbar;
+
+    @Bind(R.id.range_seekbar)
+    AppCompatSeekBar rangeSeekbar;
+
+    @Bind(R.id.search_days_ahead_label)
+    TextView searchDaysAheadLabel;
+
+    @Bind(R.id.search_range_label)
+    TextView searchRangeLabel;
+
+    @Bind(R.id.edit_account_button)
+    Button editAccountButton;
+
+    @Bind(R.id.change_password_button)
+    Button changePasswordButton;
+
+    @Bind(R.id.pref_email_text)
+    TextView emailTextView;
+
+    @Bind(R.id.pref_name_text)
+    TextView nameTextView;
+
+    @Bind(R.id.pref_signature_text)
+    TextView signatureTextView;
+
+    @Bind(R.id.pref_last_name_text)
+    TextView surnameTextView;
+
+    @Bind(R.id.pref_username_text)
+    TextView usernameTextView;
+
+    private PublishSubject<Boolean> enableNotificationsChange = PublishSubject.create();
+    private PublishSubject<Boolean> enableLocationChange = PublishSubject.create();
+    private PublishSubject<Boolean> enableNotificationsSoundChange = PublishSubject.create();
+    private PublishSubject<Boolean> enableNotificationsVibrateChange = PublishSubject.create();
+    private PublishSubject<Integer> searchDaysChange = PublishSubject.create();
+    private PublishSubject<Integer> searchRangeChange = PublishSubject.create();
+
+    public SettingsScreen()
+    {
+        super(R.layout.settings_screen);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        App.component(this).inject(this);
+        super.onCreate(savedInstanceState);
+        setTitle(R.string.title_preferences);
+
+        initializeListeners();
+    }
+
+    private void initializeListeners()
+    {
+        notificationSwitch.setOnCheckedChangeListener(this::notificationSwitchChanged);
+        notificationSoundCheckBox.setOnCheckedChangeListener((compoundButton, b) -> enableNotificationsSoundChange.onNext(b));
+        notificationVibrateCheckBox.setOnCheckedChangeListener((compoundButton, b) -> enableNotificationsVibrateChange.onNext(b));
+        locationSwitch.setOnCheckedChangeListener((compoundButton, b) -> enableLocationChange.onNext(b));
+        daysSeekbar.setOnSeekBarChangeListener(new AppCompatSeekBar.OnSeekBarChangeListener()
+        {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int i, boolean b)
+            {
+                if (i < 1)
+                {
+                    seekBar.setProgress(1);
+                }
```

Comment:
I think this is some kind of "LOGIC" so it should be in presenter. Pls make publishSubject<Integer> here :)